var/spack/repos/builtin/packages/phist/package.py

```diff
@@ -75,13 +75,15 @@ class Phist(CMakePackage):
                         'iterative solvers. For the Trilinos backends '
                         '(kernel_lib=epetra|tpetra) we can use preconditioner '
                         'packages such as Ifpack, Ifpack2 and ML.')
-
+    variant('fortran', default=True,
+            description='generate Fortran 2003 bindings (requires Python3 and '
+                        'a Fortran compiler)')
     # ###################### Dependencies ##########################
 
     depends_on('cmake@3.8:', type='build')
     depends_on('blas')
     depends_on('lapack')
-    depends_on('python@3:', when='@1.7:', type='build')
+    depends_on('python@3:', when='@1.7: +fortran', type='build')
```

Comment:
why is the dependency now more restrictive? I think you wanted to add a new line with
`depends_on('python@3:', when='+fortran', type='build')`

Reply:
I guess the question is, is Python needed when building version 1.7+ _and_ Fortran support, or when building 1.7+ _or_ Fortran?

Reply:
In fact I only need Python@3: if the version is 1.7: *and* +fortran because +fortran won't do anything in older versions

Reply:
Previously the package required `python@3:` even without `+fortran` being present. Now `spack install phist@1.7.0 ~fortran` will NOT install `python@3:`, even though it is identical to `spack install phist@1.7.0` before this patch, which DID require `python@3:`. 
If this is an intentional change, i.e. `phist@1.7.0: ~fortran` does not require `python@3:` you should mention that.

If `+fortran` doesn't do anything for `phist@:1.6.99`, should we then add a `conflicts('+fortran', when='@:1.6.99', msg='Variant fortran only exists in @1.7.0:')`?

Reply:
I guess this issue resolved? I see this code is added below:

```
conflicts('+fortran', when='@:1.6.99')
```